src/platform/ASR/ASRFactoryDataParser.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "stdint.h"
+#include "string.h"
+#include "tlv_decode.h"
+#ifdef CFG_PLF_RV32
+#include "asr_flash.h"
+#include "asr_rv32.h"
+#elif defined CFG_PLF_DUET
+#include "duet_cm4.h"
+#include "duet_flash.h"
+#define asr_flash_read duet_flash_read
+#define asr_factory_decrypt_dac_prvkey duet_factory_decrypt_dac_prvkey
+#else
+#include "lega_cm4.h"
+#include "lega_flash.h"
+#define asr_flash_read lega_flash_read
+#endif
+
+#ifndef CONFIG_ENABLE_ASR_FACTORY_DATA_PROVIDER
+#define CONFIG_ENABLE_ASR_FACTORY_DATA_PROVIDER 0
+#endif // CONFIG_ENABLE_ASR_FACTORY_DATA_PROVIDER
+
+#ifndef CONFIG_ENABLE_ASR_FACTORY_DEVICE_INFO_PROVIDER
+#define CONFIG_ENABLE_ASR_FACTORY_DEVICE_INFO_PROVIDER 0
+#endif // CONFIG_ENABLE_ASR_FACTORY_DEVICE_INFO_PROVIDER
+
+#define ASR_CONFIG_BASE PARTITION_MATTER_CONFIG
+#define ASR_MATTER_FACTORY_VERSION 2
+#define ASR_CONFIG_MATTER_NO_KEY 0x01
+
+#if CONFIG_ENABLE_ASR_FACTORY_DATA_PROVIDER
+typedef enum ASR_MATTER_PARTITION_T
```

Comment:
In c++ code it's better to use `enum class`, IMHO. Same for `factory_error_t`.

Reply:
"typedef enum" is allowed, I don't think it needs to be changed here.
In our partition , It will use as an integral constant-expression